nighthawk/include/nighthawk/client/client_worker.h

```diff
@@ -0,0 +1,43 @@
+#pragma once
+
+#include <memory>
+
+#include "envoy/common/pure.h"
+#include "envoy/stats/store.h"
+
+#include "nighthawk/client/benchmark_client.h"
+#include "nighthawk/common/statistic.h"
+#include "nighthawk/common/worker.h"
+
+namespace Nighthawk {
+namespace Client {
+
+/**
+ * Interface for a threaded benchmark client worker.
+ */
+class ClientWorker : virtual public Worker {
+public:
+  /**
+   * Gets the statistics, keyed by id.
+   * @return StatisticPtrMap A map of Statistics keyed by id.
+   */
+  virtual StatisticPtrMap statistics() const PURE;
+
+  /**
+   * @return const Envoy::Stats::Store& the statistics store associated the benchmark client.
+   */
+  virtual Envoy::Stats::Store& store() const PURE;
```

Comment:
I guess a fundamental question is why we have to expose both a store and statistics.. I think this is because the `StatisticPtrMap` only handle very specialized statistics, rather than general "stats".